src/sensai/vector_model.py

```diff
@@ -33,13 +34,44 @@ def predict(self, x: pd.DataFrame) -> pd.DataFrame:
         pass
 
     @abstractmethod
-    def getPredictedVariableNames(self):
+    def isRegressionModel(self) -> bool:
         pass
 
     @abstractmethod
-    def isRegressionModel(self) -> bool:
+    def getPredictedVariableNames(self) -> list:
         pass
 
+    def fitPreprocessors(self, X: pd.DataFrame, Y: pd.DataFrame = None):
+        # no need for fitGenerate if chain is empty
+        if len(self._inputTransformerChain) == 0 and self._featureGenerator is not None:
```

Comment:
There was a mistake in RuleBased model before, where the chain was not fitted on the generator's output. The new method fixes it. Depending on the nature of the feature generator, treating the common case of trivial DFTchains separately can result in a significant improvement in performance